TestApps/Samples/Samples/TreeViewEvents.cs

```diff
@@ -125,9 +136,15 @@ protected override Size OnGetRequiredSize ()
 				textSize = layout.GetSize ();
 			}
 
-			status.LastCalculatedHeight = textSize.Height;
+			//in cases when there are multiple lines and they don't execeed the max width we need to care height include the expand
+			double height = textSize.Height;
+			if (!status.Expanded && textSize.Height > defaultHeight) {
```

Comment:
This can be simplified. If there are multiple lines, just measure the height of the first line.